sdk/typespec/typespec_client_core/src/sleep.rs

```diff
@@ -27,3 +31,13 @@ use crate::{async_runtime::get_async_runtime, time::Duration};
 pub async fn sleep(duration: Duration) {
     get_async_runtime().sleep(duration).await
 }
+
+#[cfg(target_family = "wasm")]
+pub async fn sleep(duration: Duration) {
```

Comment:
Rather than special casing sleep, why not build a wasm specific async runtime which uses `gloo_timers` - that would be consistent with the other implementations, and be more consistent with the design. 

Bypassing the async runtime in this fashion means that on wasm, you prevent customers from providing their own sleep implementation that *isn't* gloo_timers based, which is counter to the core design.

Reply:
You're right @LarryOsterman.. That was a quick hack to prove it works. I've submitted new commits to revert that change, and introduced a web_runtime to better align with the design of the crate. Please take another look! Thx!